log-collector-script/EKS-ClusterCrawler.sh

```diff
@@ -0,0 +1,129 @@
+# #!/bin/bash
+
+ROOT_OUTPUT_DIR=$PWD
+echo "======================================="
+echo "Creating a Folder at:  $ROOT_OUTPUT_DIR"
+echo "======================================="
+kubectl config current-context > clusterName.txt
+CLUSTERNAME=$(sed 's/^[^=]*://' clusterName.txt)
+TIME=$(date "+%Y%m%d-%Hh:%Mm:%Ss")
+OUTPUT_DIR_NAME=$(sed 's|r/|r-|g' <<< "${CLUSTERNAME}").$TIME
+
+
+OUTPUT_DIR="${OUTPUT_DIR_NAME}"
+EXTENSION=${2:-'log'}
+echo "======================================="
+echo " ${CLUSTERNAME} Log Collected In Folder :  $OUTPUT_DIR"
+echo "======================================="
+cd $ROOT_OUTPUT_DIR
+mkdir "$OUTPUT_DIR"
+echo "Collecting information about kubernetes cluster: ${CLUSTERNAME} "
+FILENAME="${OUTPUT_DIR}/ClusterDetails.txt"
+kubectl config current-context > "$FILENAME"
+echo "==========================" >> "$FILENAME"
+kubectl cluster-info >> "$FILENAME"
+echo "==========================" >> "$FILENAME"
+
+
+# Collecting All the PODs descriptions/logs running in User Desired Namespace Or by default it will collect pods log running in default namespace
+Default_Namespace=${1:-'default'}
+kubectl get ns --no-headers | while read -r line; do
+NAMESPACE=$(echo "$line" | awk '{print $1}')
+if [[ "$NAMESPACE" = "$Default_Namespace" ]];then
+    kubectl get pods -n "$NAMESPACE" --no-headers | while read -r lines; do
+            POD_NAME=$(echo "$lines" | awk '{print $1}')
+            echo "$POD_NAME" 
+            FILENAME1="${OUTPUT_DIR}/${NAMESPACE}.${POD_NAME}.describe.txt"
+            kubectl describe pod -n "$NAMESPACE" "$POD_NAME" > "$FILENAME1"
+            for CONTAINER in $(kubectl get po -n "$NAMESPACE" "$POD_NAME" -o jsonpath="{.spec.containers[*].name}"); do
+            FILENAME_PREFIX="${OUTPUT_DIR}/${NAMESPACE}.${POD_NAME}.${CONTAINER}"
```

Comment:
We are looking each and every container in the cluster. Have we load tested this script?